geopandas/vectorized.pyx

```diff
@@ -0,0 +1,1029 @@
+# distutils: sources = geopandas/algos.c
+
+import collections
+import numbers
+from libc.stdlib cimport free as cfree
+
+
+import shapely
+from shapely.geometry import MultiPoint, MultiLineString, MultiPolygon
+from shapely.geometry.base import BaseGeometry, geom_factory
+from shapely.ops import cascaded_union, unary_union
+import shapely.affinity as affinity
+
+import cython
+cimport cpython.array
+from libc.stdlib cimport malloc, free
+
+
+cimport numpy as np
+import numpy as np
+
+include "_geos.pxi"
+
+from shapely.geometry.base import (GEOMETRY_TYPES as GEOMETRY_NAMES, CAP_STYLE,
+        JOIN_STYLE)
+
+cdef extern from "algos.h":
+    ctypedef char (*GEOSPredicate)(GEOSContextHandle_t handler,
+                                   const GEOSGeometry *left,
+                                   const GEOSGeometry *right) nogil
+    ctypedef char (*GEOSPreparedPredicate)(GEOSContextHandle_t handler,
+                                           const GEOSPreparedGeometry *left,
+                                           const GEOSGeometry *right) nogil
```

Comment:
I've made these definitions rather than the GEOS library, so I kept them separate. We could put them in there though, or in a separate file.